scripts/cardboardlinter

```diff
@@ -132,18 +148,51 @@ def run_diff(refspec_parent):
         that allows efficient testing in the `indiff` method of the Message class. When
         line_numbers is None, it means that all lines should be considered for testing.
     """
-    # generate 0 lines of context (i.e. only the lines that differ)
     if refspec_parent is not None:
         # The option --relative is used to limit the output of git diff to the current
         # directory, to facilitate test runs in subdirectories.
+        # -U0: generate 0 lines of context (i.e. only the lines that differ)
         command = ['git', 'diff', '-U0', refspec_parent, '--relative']
         diff_output = run_command(command)[0]
-        return parse_diff(diff_output)
+        files_lines = parse_diff(diff_output)
     else:
-        # Just get the current list of files in the repo and include all lines
-        command = ['git', 'ls-files']
+        # Just get the current list of files in the repo, including
+        # untracked files, and include all lines
+        command = ['git', 'ls-files', '--exclude-standard']
+        ls_output = run_command(command)[0]
+        files_lines = {filename: None for filename in ls_output.splitlines()}
+        command = ['git', 'ls-files', '--others', '--exclude-standard']
         ls_output = run_command(command)[0]
-        return {filename: None for filename in ls_output.splitlines()}
+        files_lines.update({filename: None for filename in ls_output.splitlines()})
+    return files_lines
+
+
+def check_encoding(files_lines):
```

Comment:
I'm not 100% sure, but `files_lines` seems to include all files, not just python files. Will this be a problem for non-python files?

Reply:
It would be a problem for binary files, but linting is typically only done in text files. I don't know of any linter for binary files in general

Reply:
OK, I see what you mean now: should we expect non-python text files to be UTF-8 as well? The alternative would be that we have to guess the encoding when reading files, which is not 100% possible. (See https://stackoverflow.com/questions/436220/how-to-determine-the-encoding-of-text .) In practice, I expect text files in our projects to always be either UTF-8 or ASCII (which is a subset of UTF-8).

Reply:
I'm taking out the encoding checks, because I'm not entirely happy with it. Needs to be done better.